src/util/util.c

```diff
@@ -1091,3 +1091,81 @@ errno_t sss_getenv(TALLOC_CTX *mem_ctx,
 
     return value != NULL ? EOK : ENOENT;
 }
+
+errno_t sss_parse_uri(TALLOC_CTX *mem_ctx, const char *uri, struct sss_parsed_uri **_parsed_uri)
```

Comment:
Maybe rename this to sss_parse_dns_uri? This is not usable by common uris.

Also, you can use libcurl to parse it for you https://curl.se/libcurl/c/libcurl-url.html CURLUPART_FRAGMENT will be the DoT server hostname.

Reply:
Renamed. It can actually parse ordinary URL but the DNS URI is special.

Unfortunately standard libraries like CURL or libldap (this also has URI parsing function) fail on DNS URIs, so we have to parse it on our own.

Reply:
Ok, that's unfortunate.

It can "parse" them, but content after `#` is treated as hostname instead of uri fragment. That's why I think it is only for this kind of DNS URI.